linked_list.py

```diff
@@ -7,32 +8,33 @@ def __init__(self, data, nextNode=None):
         self.nextNode = nextNode
 
     def __str__(self):
-        return self.data
+        return str(self.data)
```

Comment:
You might want to provide some way of differentiating a node that contains a value from the value itself.  But this is a vast improvement over what you had before.  Good. (same goes for `__repr__` below)